libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -117,11 +116,44 @@ proc handle*(p: PubSubPeer, conn: Connection) {.async.} =
       debug "exiting pubsub peer read loop"
       await conn.close()
 
+      if p.sendConn == conn:
+        p.sendConn = nil
+
   except CancelledError as exc:
     raise exc
   except CatchableError as exc:
     trace "Exception occurred in PubSubPeer.handle", exc = exc.msg
 
+proc getSendConn(p: PubSubPeer): Future[Connection] {.async.} =
+  # get a cached send connection or create a new one - need to call
+  # `putSendConn` after use! If one is not available, set up a new one
+  var res = p.sendConn
+
+  if not res.isNil:
+    if res.closed() or res.atEof:
```

Comment:
Wouldn't this code be more natural to read with an early return?

```nim
if res != nil:
  if not (res.closed or res.atEof):
    return res
  p.sendConn = nil
  await res.close() # p.sendConn might change after here
 
res = await p.switch.dial(...)
```

It's a cache after all, so an early return is a natural thing to have.

Reply:
changed it, but I'm not sure the logic is any more simple to follow